src/compile/mark/valueref.ts

```diff
@@ -83,18 +75,22 @@ export function midPoint(channel: Channel, channelDef: ChannelDef, scaleName: st
   if (channelDef) {
     /* istanbul ignore else */
     if (isFieldDef(channelDef)) {
+      if (scale.type === 'bin-linear') {
+        return {
+          signal: `(scale("${scaleName}", ${field(channelDef, {binSuffix: 'start', datum: true})}) + scale("${scaleName}", ${field(channelDef, {binSuffix: 'end', datum: true})}))/2`
```

Comment:
> it only works for certain channels

Current way doesn't make it better in this aspect.  

>  making it a separate function hides the implementation.

That's better that showing complex code that's hard to read.  Plus, keeping the method provides consistent level of abstraction in this method.